it/controllers/benefits/reimbursed/VouchersOrCreditCardsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,534 @@
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.vouchersAndCreditCardsQuestion)) shouldBe Some(false)
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.vouchersAndCreditCards)) shouldBe None
+        }
+
+      }
+
+      "Update the vouchers or credit cards question to no, and vouchers or credit cards to none when the user chooses no," +
+        "redirects to next question if not prior submission" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = false, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        //          TODO: This will need updating to the next page in the flow when the navigation is hooked up
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY,employmentId).url)
+        }
+
+        "updates vouchers or credit cards question to no and vouchers or credit cards to none" in {
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.vouchersAndCreditCardsQuestion)) shouldBe Some(false)
+          cyamodel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.vouchersAndCreditCards)) shouldBe None
+        }
+
+      }
+
+      "Update the vouchers or credit cards question to yes, redirects to the vouchers or credit cards amount page" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullReimbursedCostsVouchersAndNonCashModel.copy(vouchersAndCreditCardsQuestion = Some(false)))))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the vouchers or credit cards amount page" in {
+          result.status shouldBe SEE_OTHER
+          //          TODO: This will be updated to the vouchers or credit cards amount page when its created
```

Comment:
remove when updating the controllers redirect

Reply:
This will need to be updated when the navigation for the section is completed.